### ISSUE TITLE
[Bug]: Crashing while using beds

### ISSUE THREAD

**Issue by FatCodeNerd** [TheButterbrotMan/Deathdusk#136]
### What happened?

Trying to use beds in foreign places, like those pre-fab houses or this one like camp tent just outside an MCA village. The game didn't actually crash, as in produce a crash report, it just froze and loaded infinitely.  ALSO. It was saying the bed was occupied despite the bed being empty a few times. 

### Version

2.1.1

### Provide the log

There was no crash log. It didn't produce one

**Comment by Butamichi** [comment]
was it a vanilla bed generated on a structure or a modded one made by you?

**Comment by FatCodeNerd** [comment]
> was it a vanilla bed generated on a structure or a modded one made by you?

it was a vanilla bed generated as part of a structure. a vanilla bed, not that modded bed stuff, in a pre-fab tent structure that spawned in the world

**Comment by TheButterbrotMan** [comment]
I need a latest.log please. Did this occur in single or multiplayer?

**Comment by FatCodeNerd** [comment]
Part two of bug report as per request: 
https://cdn.discordapp.com/attachments/413011160904761344/1001449776132390962/IMG_8345.jpg
https://cdn.discordapp.com/attachments/413011160904761344/1001449776434401301/IMG_8346.jpg

https://bytebin.lucko.me/16Z1oZKPjx

Images of the structure and the latest.log 

It's important to note on certain reloads of the game, walking up to the bed and attempting to right-click it would prompt the user with the message "This bed is occupied", as in implying someone was in the bed, despite it being clearly not occupied. Breaking and replanting the bed fixed this issue. In either situation, it would crash after attempting to use the bed to sleep to the next day. In the distance to the side is an MCA settlement but other then that its mostly low hills and plains around the area (a great place for building)

**Comment by FatCodeNerd** [comment]
I went to a creative world to test things out, and immediately spawned in two things

1) a default bed (from minecraft itself)
2) a multibed bed 

set gamemode to night and went in default bed. the entire game chugged and struggled to process what was happening, probably getting single digit to low double digit FPS here, but DIDN'T crash. Did the same test in a multibed bed and it was significantly less laggier but still lagged quite a bit. After BOTH tests, once night had fast forwarded to day the lag immediately ended. 

I went around to look for a village and found the same BLUE HOUSE type that simply approaching had caused to crash earlier. I don't know if it was the creative mode or what but simply entering the house didn't crash this time. Did lag however. Sleeping in a bed though, which were multibed beds, crashed the game. No crash log, but I did get the latest.log 
https://bytebin.lucko.me/thgazPGrmj

My next attempt was simply joining the original survival game I had been enjoying. I was going to attempt to replicate the glitch but a super zombie (???) that was impossible to kill at my current equipment level was essentially spawncamping me, so I was forced to rush to the MCA village nearby and sleep in one of there beds to try and make it morning. I was expecting a crash, and it froze. But I kept patience and waited it out. Eventually, I had slept, and it was the next day. No lag. The previous night I had broken and stolen a bed from the MCA village to test in the exact place of the camp bed (a minecraft original bed) that was the source of all the crashes, but now it's day time, and I don't really don't want to try and replicate the glitch the next time its night. 

I determined that while I wasn't necessarily IN the village, quite far away from it, I was still within its territorial bounds, evidence by when I walked/boated away far enough it now said 'Now Leaving Dawnstar' (or whatever the village was called)

So I don't know what this means.
